modmail/extensions/utils/error_handler.py

```diff
@@ -0,0 +1,195 @@
+        """Activates when a command raises an error."""
+        if getattr(error, "handled", False):
+            logging.debug(f"Command {ctx.command} had its error already handled locally; ignoring.")
+            return
+
+        if isinstance(error, commands.CommandNotFound):
+            # ignore every time the user inputs a message that starts with our prefix but isn't a command
+            # this will be modified in the future to support prefilled commands
+            if ANY_DEV_MODE:
+                logger.trace(error)
+            return
+
+        logger.trace(error)
+
+        title = None
+        msg = None
+        embed: typing.Optional[discord.Embed] = None
+        should_respond = True
+
+        if isinstance(error, commands.UserInputError):
+            embed = await self.handle_user_input_error(ctx, error)
+        elif isinstance(error, commands.CheckFailure):
+            embed = await self.handle_check_failure(ctx, error)
+            # handle_check_failure may send its own error if its a BotMissingPermissions error.
+            if embed is None:
+                should_respond = False
+        elif isinstance(error, commands.ConversionError):
+            # s = object()
+            # error.converter.convert.__annotations__.get("return", s)
+            # embed = error
+            ...
+        elif isinstance(error, commands.DisabledCommand):
+            logger.debug("")
+            if ctx.command.hidden:
+                should_respond = False
+            else:
+                msg = f"Command `{ctx.invoked_with}` is disabled."
+                if reason := ctx.command.extras.get("disabled_reason", None):
+                    msg += f"\nReason: {reason}"
+                embed = self.error_embed(msg, title="Command Disabled")
+
+        elif isinstance(error, commands.CommandInvokeError):
+            # generic error
+            logger.error(f'Error occurred in command "{ctx.command}".', exc_info=error.original)
+            # todo: this should log somewhere else since this is a bot bug.
+            embed = self.error_embed(
+                "Oops! Something went wrong internally in the command you were trying to execute. "
+                "Please report this error and what you were trying to do to the developers."
+            )
+            if isinstance(error.original, discord.Forbidden):
+                await self.handle_bot_missing_perms(ctx, error.original)
+                return
+
+        # TODO: this has a fundamental problem with any BotMissingPermissions error
+        # if the issue is the bot does not have permissions to send embeds or send messages...
+        # yeah, problematic.
+
+        if not should_respond:
+            logger.debug("Not responding to error since should_respond is falsey.")
+            return
+
+        if embed is None:
+            embed = self.error_embed(msg or str(error), title=title or self.get_title_from_name(error))
```

Comment:
Since neither title nor  message are actually used from the above, they can be dropped. This should also be commented, I think:
`# general case` or something like that.